src/vendor/azure-uamqp-c/src/link.c

```diff
@@ -1083,6 +1124,84 @@ int link_get_peer_max_message_size(LINK_HANDLE link, uint64_t* peer_max_message_
     return result;
 }
 
+int link_get_offered_capabilities(LINK_HANDLE link, AMQP_VALUE* offered_capabilities)
+{
+    int result;
+    if((link == NULL) ||
+        (offered_capabilities == NULL))
+    {
+        LogError("Bad arguments: link = %p, offered_capabilities = %p",
+            link, offered_capabilities);
+        result = __FAILURE__;
+    }
+    else if ((link->link_state != LINK_STATE_ATTACHED) &&
+    (link->link_state != LINK_STATE_HALF_ATTACHED_ATTACH_RECEIVED))
+    {
+                LogError("Attempting to read offered capabilities before it was received");
+        result = __FAILURE__;
+    }
+    else
+    {
+        *offered_capabilities = link->offered_capabilities;
+        result = 0;
+    }
+    return result;
+}
+
+int link_set_offered_capabilities(LINK_HANDLE link, AMQP_VALUE offered_capabilities)
+{
+    int result;
+
+    if (link == NULL)
+    {
+        LogError("NULL link");
+        result = __FAILURE__;
+    }
+    else
+    {
+        link->offered_capabilities = offered_capabilities;
```

Comment:
We need to do `amqpvalue_clone` on the input before assigning it.

Reply:
fixed